### PR TITLE
Migration to asciidoc

### PR DESCRIPTION
**Pull request by jmfinelli**
This PR addresses [JBTM-3921](https://issues.redhat.com/browse/JBTM-3921)

### CODE REVIEW

**Comment by jmfinelli** [comment]
TESTIT

**Comment by marcosgopen** [comment]
@jmfinelli stunning work with this PR!!!
I am not sure how to review all this PR. I have some considerations that are actually more fitting for https://issues.redhat.com/browse/JBTM-3914
Could we talk about those here?

**Comment by jmfinelli** [comment]
> Rather than: product/product_documentation.html and project/project_documentation.html

Done

File: src/main/asciidoc/project/jts/trailmap.adoc [hunk 1]
```diff
@@ -0,0 +1,4765 @@
...
+===== Recovery and Logging
+
+Basically, the Recovery is the mechanism which preserves the transaction atomicity in presence of failures.
+The basic technique for implementing transactions in presence of failures is based on the use of logs.
+That is, a transaction system has to record enough information to ensure that it can be able to return to a previous state in case of failure or to ensure that changes committed by a transaction are properly stored.
+
+In addition to be able to store appropriate information, all participants within a distributed transaction must log similar information, which allows them to take a same decision either to set data in their final state or in their initial state.
+
+Two techniques are, in general, used to ensure transaction's atomicity.
+A first technique focuses on manipulated data, such as the Do/Undo/Redo protocol (considered as a recovery mechanism in a centralised system), which allows a participant to set its data in their final values or to retrieve them in their initial values.
```
Comment (marcosgopen) [inline src/main/asciidoc/project/jts/trailmap.adoc:328]: This file is strangely long! I can see that it is well migrated, but I can see duplication in this new file and the old one. i.e. these lines https://github.com/jbosstm/documentation/pull/73/files#diff-62ade1ca747887147020c9aa09a6bcdf9074ceb197e58ec5cd41cc67eecb1629R278-R333 can be found also at

**Comment by marcosgopen** [comment]
Great work with the migration @jmfinelli ! LGTM +1

**Review by marcosgopen — APPROVED** [review]

File: src/main/asciidoc/project/jts/trailmap.adoc (hunk 1 above)
Comment (jmfinelli) [reply src/main/asciidoc/project/jts/trailmap.adoc:328]: Thanks @marcosgopen. I also noticed that `trailmap.adoc` has some duplications (and other problems), which are also present in the current documentation. My suggestion is to go ahead with the review and then fix `trailmap.adoc` once we merge this PR. WDYT?

**Pull request merged**